src/kit/shell/src/shellWindows.c

```diff
@@ -274,32 +276,35 @@ int32_t shellReadCommand(TAOS *con, char command[]) {
   // Read input.
   void *console = GetStdHandle(STD_INPUT_HANDLE);
   unsigned long read;
-  wchar_t c;
+  wchar_t *c= (wchar_t *)calloc(SHELL_INPUT_MAX_COMMAND_SIZE, sizeof(wchar_t));
   char mbStr[16];
   while (1) {
-    int ret = ReadConsole(console, &c, 1, &read, NULL);
-    int size = WideCharToMultiByte(CP_UTF8, 0, &c, read, mbStr, sizeof(mbStr), NULL, NULL);
-    mbStr[size] = 0;
-    switch (c) {
-      case '\n':
-        if (isReadyGo(&cmd)) {
-          sprintf(command, "%s%s", cmd.buffer, cmd.command);
-          free(cmd.buffer);
-          cmd.buffer = NULL;
-          free(cmd.command);
-          cmd.command = NULL;
-          return 0;
-        } else {
-          shellPrintContinuePrompt();
-          updateBuffer(&cmd);
-        }
-        break;
-      case '\r':
-        break;
-      default:
-        for (int i = 0; i < size; ++i) {
-          insertChar(&cmd, mbStr[i]);
-        }
+    int ret = ReadConsole(console, c, SHELL_INPUT_MAX_COMMAND_SIZE, &read, NULL);
+    for (int input_index = 0; input_index < read; input_index++) {
+      int size = WideCharToMultiByte(CP_UTF8, 0, &c[input_index], 1, mbStr, sizeof(mbStr), NULL, NULL);
+      mbStr[size] = 0;
+      switch (c[input_index]) {
+        case '\n':
+          if (isReadyGo(&cmd)) {
+            sprintf(command, "%s%s", cmd.buffer, cmd.command);
+            free(cmd.buffer);
+            cmd.buffer = NULL;
+            free(cmd.command);
+            cmd.command = NULL;
+            free(c);
+            return 0;
+          } else {
+            shellPrintContinuePrompt();
+            updateBuffer(&cmd);
+          }
+          break;
+        case '\r':
```

Comment:
array 'c' be freed in case '\n' only, but seems the memory is not freed if the loop break from 'r'?